src/App.js

```diff
@@ -1,79 +1,43 @@
-import React from 'react'
-import { Experiment, jsPsych } from 'jspsych-react'
-import { tl } from './timelines/main'
-import { MTURK, FIREBASE } from './config/main'
-import './App.css'
-import 'bootstrap/dist/css/bootstrap.css'
-import '@fortawesome/fontawesome-free/css/all.css'
-import { getTurkUniqueId, sleep } from './lib/utils'
-import { rt_categorize_html } from './lib/rt-categorize-html'
-import { addToFirebase, createFirebaseDocument,} from "./firebase.js";
+import React, { useState, useEffect } from "react";
+import "./App.css";
+import "bootstrap/dist/css/bootstrap.css";
+import "@fortawesome/fontawesome-free/css/all.css";
+import Login from "./components/Login";
+import JsPsychExperiment from "./components/JsPsychExperiment";
+import { MTURK, IS_ELECTRON } from "./config/main";
+import { getTurkUniqueId } from "./lib/utils";
+import LoginContext from "./contexts/LoginContext";
 
+function App() {
+  // Variables for login context
+  const [loggedIn, setLogin] = useState(false);
+  const login = () => setLogin(true);
 
+  const [ipcRenderer, setRenderer] = useState(false);
+  const [psiturk, setPsiturk] = useState(false);
 
+  // All Mturk and other login logic will go here
+  useEffect(() => {
+    if (IS_ELECTRON) {
+      const electron = window.require("electron");
+      setRenderer(electron.ipcRenderer);
+    } else if (MTURK) {
```

Comment:
the old login trial also (I believe) had a way to get login info from the environment if electron - could you check into that and add here as well?

Reply:
I think its this line `const envPatientId = ipcRenderer.sendSync('syncPatientId')`. I'm not entirely sure what this sendSync does and it doesn't work if I just directly copy the code over.

Reply:
@mcmcgrath13 Ok I got it to work now by editing the electron.js file. It should autofil the login fields if the environment variables for patientId and studyId were set.

Reply:
where are you sending the sync to get the env vars?

Reply:
That is in login.js since I'm just using it to autofill the fields.